test/test-cases/functional/saic/test_vm_to_vm_commn_udp_bidir.py

```diff
@@ -0,0 +1,295 @@
+        udp.dst_port.value = 4789
+
+        #vxlan.flags.value = 
+        vxlan.vni.value = 11
+        vxlan.reserved0.value = 0
+        vxlan.reserved1.value = 0
+
+        inner_eth.src.value = INNER_SRC_MAC
+        inner_eth.dst.value = INNER_DST_MAC2
+
+        
+        inner_ip.src.value = ENI_IP    #ENI
+        inner_ip.dst.value = NETWORK_IP2  #world
+
+        inner_udp.src_port.value = 10000
+        inner_udp.dst_port.value = 20000
+
+        print("\n======= Configure Flow4 from NETWORK2 to ENI =======")        
+        # Flow4 settings
+        f4 = dataplane.configuration.flows.flow(name="NETWORK2_TO_ENI")[-1]
+        f4.tx_rx.port.tx_name = dataplane.configuration.ports[1].name
+        f4.tx_rx.port.rx_name = dataplane.configuration.ports[0].name
+        f4.size.fixed = PACKET_LENGTH
+        # send n packets and stop
+        f4.duration.fixed_packets.packets = TOTALPACKETS
+        # send n packets per second
+        f4.rate.pps = PPS
+        f4.metrics.enable = True
+
+        outer_eth, ip, udp, vxlan, inner_eth, inner_ip , inner_udp= (
+                f4.packet.ethernet().ipv4().udp().vxlan().ethernet().ipv4().udp()
+        )
+        
+        outer_eth.src.value = OUTER_SRC_MAC_F2
+        outer_eth.dst.value = OUTER_DST_MAC_F2 
+        outer_eth.ether_type.value = 2048
+
+        ip.src.value = NETWORK_VTEP_IP
+        ip.dst.value = DPU_VTEP_IP
+
+        udp.src_port.value = 11638
+        udp.dst_port.value = 4789
+
+        #vxlan.flags.value = 
+        vxlan.vni.value = 101
+        vxlan.reserved0.value = 0
+        vxlan.reserved1.value = 0
+
+        inner_eth.src.value = INNER_DST_MAC2
+        inner_eth.dst.value = INNER_SRC_MAC
+
+        inner_ip.src.value = NETWORK_IP2 #world
+        inner_ip.dst.value = ENI_IP   # ENI
+
+        inner_udp.src_port.value = 20000
+        inner_udp.dst_port.value = 10000
+        dataplane.set_config()
+
+        # Verify Traffic
+        print("\n======= Start Traffic  =======")        
+        su.start_traffic(dataplane, f1.name)
+        su.start_traffic(dataplane, f3.name)
+        time.sleep(0.5)
+        su.start_traffic(dataplane, f2.name)
+        su.start_traffic(dataplane, f4.name)
+        time.sleep(TRAFFIC_SLEEP_TIME)
+        print("\n======= Stop Traffic  =======")                    
+        dataplane.stop_traffic()
+
+        print("\n======= Verify Traffic flows   =======")        
+        res1 = su.check_flow_tx_rx_frames_stats(dataplane, f1.name)
+        res2 = su.check_flow_tx_rx_frames_stats(dataplane, f2.name)
+        res3 = su.check_flow_tx_rx_frames_stats(dataplane, f3.name)
+        res4 = su.check_flow_tx_rx_frames_stats(dataplane, f4.name)
+        
+        dataplane.teardown()
+
+        # Validate test result  
+        print("\n======= Print Test Results  =======")        
+        print("res1 and res2 and res3 and res4 is {} {} {} {}".format(res1, res2, res3, res4))
+        assert res1, "Traffic test failure"
```

Comment:
It would be good if you will identify which traffic item failed - endpoints/direction/in or outbound
Also you can build and print a table that can help to debug (see prettytable module).

Reply:
if we use table modules, that is  to be installed in pipeline.
Added detailed print statement instead

Reply:
thanks!